wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -1097,9 +1114,12 @@ private class WasmExpressionBuilder private (
               )
         }
 
-      case IRTypes.ArrayType(_) =>
-        println(tree)
-        ???
+      case IRTypes.ArrayType(arrayTypeRef) =>
+        /* TODO This is not correct for reference array types. It treats
```

Comment:
So, the problem is, `isInstanceOf` for reference array types will be always true because they are treated as `Array[j.l.Object]` at runtime, can we compare the `componentType` of the given instance's typeData?

Reply:
We should indeed compare the `componentType`s in *some* way, but it is more tricky than just comparing for equality, because of covariance. So it's a bit hairy to implement at this stage. We'll need enough run-time data to check whether any given `ClassRef` (represented by an instance of `typeData`) was a *subclass* of another `ClassRef`.

We will need the same run-time support to implement `jl.Class.isAssignableFrom`, btw.

Reply:
> , but it is more tricky than just comparing for equality, because of covariance.

yeah, that makes sense. Thanks!